### ISSUE TITLE
WSL Extension v0.80.0 appears to break WSL connection on VSCode Insiders

### ISSUE THREAD

**Issue by cquick01** [microsoft/vscode-remote-release#8643]
WSL Extension v0.80.0 appears to break connection with WSL. I was working (connected to WSL from Windows) then all of a sudden VSCode froze and when attempting to reopen, a message popped up saying "Windows Subsystem for Linux not detected. The WSL extension requires Windows 10 with the Windows Subsystem for Linux feature installed.". I was already connected when it crashed, and WSL still works fine outside of VSCode.

Reverting back to WSL v0.79.5 gets things working again. Updating back to v0.80.0 causes it to fail. Rebooting does not help (which would have restarted WSL, per #8642).
Edit: for completeness, also tried running `wsl --shutdown` as recommended in that issue, and restarting the WSL instance. Still fails with v0.80.0

- VSCode Version: 1.80.0-insider (user setup)
  - Commit: b4952d14a465572e33f0635ef58de089227b8876
  - Date: 2023-06-26T05:32:56.741Z
  - Electron: 22.3.14
  - ElectronBuildId: 21893604
  - Chromium: 108.0.5359.215
  - Node.js: 16.17.1
  - V8: 10.8.168.25-electron.0
  - OS: Windows_NT x64 10.0.19045
- Local OS Version: Windows 10 22H2 (19045.3086)
- Remote OS Version: Ubuntu 22.04.2 LTS
- Remote Extension/Connection Type: WSL2

`wsl --version`:

- WSL version: 1.2.5.0
- Kernel version: 5.15.90.1
- WSLg version: 1.0.51
- MSRDC version: 1.2.3770
- Direct3D version: 1.608.2-61064218
- DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
- Windows version: 10.0.19045.3086

- Logs: From `AppData/Roaming/Code - Insiders/logs/20230626T160143/window2/renderer.log`

```
140:2023-06-26 16:05:24.126 [error] [LocalProcess0][resolveAuthority(wsl,1)][109650ms] returned an error {"code":"NotAvailable","message":"WSL not installed.","detail":true}
141:2023-06-26 16:05:24.127 [error] resolveAuthority(wsl) returned an error after 109650 ms WSL not installed.
143:2023-06-26 16:05:24.127 [error] CodeExpectedError: WSL not installed.
149:2023-06-26 16:05:24.128 [error] CodeExpectedError: WSL not installed.
155:2023-06-26 16:05:24.131 [error] CodeExpectedError: WSL not installed.
163:2023-06-26 16:05:24.250 [error] CodeExpectedError: WSL not installed.
169:2023-06-26 16:05:24.315 [error] CodeExpectedError: WSL not installed.
175:2023-06-26 16:05:24.316 [error] CodeExpectedError: WSL not installed.
180:2023-06-26 16:05:24.317 [error] CodeExpectedError: WSL not installed.
185:2023-06-26 16:05:24.317 [error] Cannot register remote filesystem provider. Error while fetching remote environment. WSL not installed.
186:2023-06-26 16:05:24.318 [error] CodeExpectedError: WSL not installed.
191:2023-06-26 16:05:24.318 [error] CodeExpectedError: WSL not installed.
196:2023-06-26 16:05:24.318 [error] CodeExpectedError: WSL not installed.
201:2023-06-26 16:05:24.319 [error] CodeExpectedError: WSL not installed.
206:2023-06-26 16:05:24.327 [error] CodeExpectedError: WSL not installed.
269:2023-06-26 16:05:24.389 [error] WSL not installed.: CodeExpectedError: WSL not installed.
280:2023-06-26 16:05:24.693 [error] WSL not installed.: CodeExpectedError: WSL not installed.
285:2023-06-26 16:05:24.848 [error] CodeExpectedError: WSL not installed.
290:2023-06-26 16:05:26.296 [error] WSL not installed.: CodeExpectedError: WSL not installed.
295:2023-06-26 16:09:01.827 [error] WSL not installed.: CodeExpectedError: WSL not installed.
```

But of course WSL is installed and actively running.

Can't really find anything else other than the error message: "Windows Subsystem for Linux not detected. The WSL extension requires Windows 10 with the Windows Subsystem for Linux feature installed."

Steps to Reproduce:

1. Try connecting to WSL instance with WSL extension v0.80.0. see it fail to connect
2. Revert to v0.79.5, works again.

Does this issue occur when you try this locally?: N/A
Does this issue occur when you try this locally and all extensions are disabled?: N/A

Happy to provide more info if requested. Thanks!

**Comment by charlottia** [comment]
Thanks for raising this issue! I arrived here via #8642 and likewise a WSL shutdown didn't help — reverting to 0.79.5 did.

**Comment by phildaniels** [comment]
I have a local fix you can do in the extension code itself, I think someone forgot to return here. It's all obfuscated code so unsure  what some of this does exactly. The local fix will have to be applied after every update to the extension

![image](https://github.com/microsoft/vscode-remote-release/assets/11820265/d917da3b-b8f2-46e9-b341-57f30d9e3cbc)

**Comment by chrmarti** [comment]
This appears to be fixed in v0.80.2. Thanks. (https://github.com/microsoft/vscode-remote-wsl/pull/150)
